dvc/lock.py

```diff
@@ -44,7 +44,7 @@ def _do_lock(self):
         try:
             with Tqdm(
                 bar_format="{desc}",
-                disable=not self._friendly,
+                disable=True if not self._friendly else None,
```

Comment:
@efiop hope this change is compatible with what you intended by this line

Reply:
@casperdcl btw, are there any reasons why Tqdm requires `None` and `False` is not enough?

Reply:
A bit weird though (all of these True if ... else None) and really easy to mess up in the future. Maybe `Tqdm` from `progress.py` should ignore `disable` and always disable pbar when we are in quiet mode?

Reply:
Well we could treat `disable=False` as we currently treat `disable=None`. That would require fewer changes but would be counterintuitive to me.

Reply:
Also there are only a few cases where we need to override `None` to `True`...

Reply:
I will say we should probably drop `no_progress_bar` in a future PR

Reply:
@casperdcl Not sure I follow. I does seem to me that if we are in quiet mode, `disable=False` or `disable=None` should not print the progress bar, because loglevel logic has a higher priority over `disable` option.

Reply:
Yes. Ideally, loglevel logic *should* have always controlled verbosity, including for progress bars.

This has been the case with `disable=None`.

However there are 2 exceptions. In both cases we want to disable even if `log.level == INFO`. This requires `disable=True if condition else None`.

If in future we ever come up with a reason to use `disable=False` then we can think about how to support it at that point. I'm even tempted to (for now) include an `assert disable is not False` statement.

Reply:
@casperdcl Again, might be missing something, but why not replace:
```
 if disable is None:                               
     disable = logger.getEffectiveLevel() > level  
```
with
```
 if not disable:
     disable = logger.getEffectiveLevel() > level  
```
So if it is disabled then it is 100% disabled, but if it is not disabled then we will determine based on loglevel whether or not we actually need to enable it.

Reply:
> Well we could treat `disable=False` as we currently treat `disable=None`. That would require fewer changes but would be counterintuitive to me.

Reply:
exactly what I was suggesting but it seems a little bad programming practice to treat what is essentially an explicit `disable=False` as if it was `disable=None` (i.e. auto)

Reply:
@casperdcl Sounds alright to me because loglevel has higher priority over `disable=False`.

Reply:
@casperdcl Also, we need a test for this PR.

Reply:
right will also add test